Sources/SwiftSyntaxBuilder/StringLiteralExprConvenienceInitializers.swift

```diff
@@ -13,13 +13,33 @@
 import SwiftSyntax
 
 extension StringLiteralExpr {
-  public init(_ value: String, openQuote: TokenSyntax = .stringQuote, closeQuote: TokenSyntax = .stringQuote) {
+  /// Creates a string literal, optionally specifying quotes and delimiters.
+  public init(
+    openDelimiter: TokenSyntax? = nil,
+    openQuote: TokenSyntax = .stringQuote,
+    _ value: String,
+    closeQuote: TokenSyntax = .stringQuote,
+    closeDelimiter: TokenSyntax? = nil
+  ) {
     let content = TokenSyntax.stringSegment(value)
     let segment = StringSegment(content: content)
     let segments = StringLiteralSegments([segment])
 
-    self.init(openQuote: openQuote,
-              segments: segments,
-              closeQuote: closeQuote)
+    self.init(
+      openDelimiter: openDelimiter,
+      openQuote: openQuote,
+      segments: segments,
+      closeQuote: closeQuote,
+      closeDelimiter: closeDelimiter
+    )
+  }
+
+  /// Creates a raw string literal.
+  public init(raw value: String) {
```

Comment:
What do you think about adding an additional parameter that describes how many `#` should be added and defaulting it to 1? That way you could create a string literal like `##" The raw string delimiter is #". "##`

Reply:
Sounds good. Perhaps we could take it even further and scan the string for the longest sequence of consecutive hashes + adjacent quote and do that automatically. Or would that be too much magic?

Reply:
Sure, you might also want to check for the longest sequence of `####\(` which can also be escaped using raw strings.